_posts/2020-07-31-fnn-vae-for-noisy-timeseries/fnn_vae_for_noisy_timeseries.Rmd

```diff
@@ -0,0 +1,509 @@
+---
+title: "FNN-VAE for noisy time series forecasting"
+description: > 
+ In this last part of a mini-series on forecasting with false nearest neighbors (FNN) loss, we replace the LSTM autoencoder from the previous post by a convolutional VAE, resulting in equivalent prediction performance but significantly lower training time. In addition, we find that FNN regularization is of great help when an underlying deterministic process is obscured by substantial noise.
+  
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanafnnvae
+bibliography: bibliography.bib
+date: 07-31-2020
+categories:
+  - R
+  - TensorFlow/Keras
+  - Time Series
+  - Unsupervised Learning
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/kb.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+This post did not quite end up the way I'd imagined. A quick follow-up on the recent [Time series prediction with
+FNN-LSTM](https://blogs.rstudio.com/ai/posts/2020-07-20-fnn-lstm/), it should have shown how *noisy time series* (so common in
+practice) could profit from a change in architecture: Instead of FNN-LSTM, an LSTM autoencoder regularized by false nearest
+neighbors (FNN) loss, use FNN-VAE, a variational autoencoder constrained by the same. However, FNN-VAE did not seem to handle
+noise better than FNN-LSTM. No plot, no post, then?
+
+On the other hand -- this is not a scientific study, with hypothesis and experimental setup all preregistered; all that really
+matters is if there's something useful to report. And it looks like there is.
+
+Firstly, FNN-VAE, while on par, performance-wise, with FNN-LSTM, it is far superior in that other meaning of "performance":
+Training goes a *lot* faster for FNN-VAE.
+
+And secondly, while we don't see much difference between FNN-LSTM and FNN-VAE, we *do* see a clear impact of using FNN loss:
+Especially for the VAE architecture, but for the LSTM one as well, adding in FNN loss strongly reduces mean squared error with
```

Comment:
thanks for the link, I'll take a look! I do tend to think by now that really there be something of a usage difference between languages here: (hehe)

> I would break it into two sentences, 

For me, phrases separated by colons nearly always ARE separate sentences! Will change; and put more attention to it in the future (when reading English prose, I mean)